<?xml version="1.0"?>
<robot name="two_joint_hw_simple" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <link name="base_link"/><link name="link_hip"/><link name="link_knee"/>
  <joint name="hip" type="continuous"><parent link="base_link"/><child link="link_hip"/><axis xyz="0 0 1"/></joint>
  <joint name="knee" type="continuous"><parent link="link_hip"/><child link="link_knee"/><axis xyz="0 0 1"/></joint>

  <ros2_control name="RobstrideSystem" type="system">
    <hardware>
      <plugin>motor_control::RobstrideSystemHardwareSimple</plugin>
      <param name="can_rx_topic">/from_can_bus</param>
      <param name="can_tx_topic">/to_can_bus</param>
      <param name="motor_ids">0x01,0x02</param> 
      <param name="joint_signs">1,-1</param>     
      <param name="gear_ratios">1, 1</param>    
      <param name="vel_max">10.0</param>         
      <param name="acc_set">10.0</param>       
      <param name="can_timeout_counts">20000</param>
    </hardware>

    <joint name="hip">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="knee">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>
</robot>
